java/code/src/com/redhat/rhn/manager/action/ActionManager.java

```diff
@@ -2344,4 +2367,86 @@ public static ImageInspectAction scheduleImageInspect(User scheduler, List<Long>
+            sysids.removeAll(visible);
+            throw new UnsupportedOperationException("Some System not available or not managed with Salt: " + sysids);
+        }
+
+        List<MinionServer> minions = sysids.stream().map(
+            id -> SystemManager.lookupByIdAndUser(id, loggedInUser).asMinionServer().get()).collect(toList());
+
+        List<Long> proxies = minions.stream().filter(m -> m.isProxy()).map(m -> m.getId()).collect(toList());
+        if (!proxies.isEmpty()) {
+            throw new UnsupportedOperationException("Some of the minions are proxies: " + proxies);
+        }
+
+        Optional<Server> proxy = Optional.empty();
+
+        if (proxyId != 0) {
+            proxy = Optional.of(SystemManager.lookupByIdAndUser(proxyId, loggedInUser));
+            proxy.ifPresent(p -> {
+                if (!p.isProxy()) {
+                    throw new UnsupportedOperationException("The system is not a proxy: " + p.getId());
+                }
+            });
+        }
+        Optional<Long> proxyIdOpt = proxy.map(p -> p.getId());
+
+        List<Long> sshIds = minions.stream()
+                            .filter(minion -> ContactMethodUtil.isSSHPushContactMethod(minion.getContactMethod()))
+                            .map(minion -> {
+            // handle SSH minions
+            minion.updateServerPaths(proxyIdOpt);
+            ServerFactory.save(minion);
+
+            MinionPillarManager.INSTANCE.generatePillar(minion);
+            return minion.getId();
+        }).collect(toList());
+
+
+        List<Long> normalIds = minions.stream()
+                               .filter(minion -> !ContactMethodUtil.isSSHPushContactMethod(minion.getContactMethod()))
+                               .map(minion -> minion.getId())
+                               .collect(toList());
+
+        List<Long> ret = new ArrayList<Long>();
+        if (!sshIds.isEmpty()) {
+            // action for SSH minions - update channel configuration
+            Action a = scheduleApplyStates(loggedInUser, sshIds,
+                       Collections.singletonList(ApplyStatesEventMessage.CHANNELS),
+                       new Date());
+            a = ActionFactory.save(a);
+            taskomaticApi.scheduleActionExecution(a);
+            ret.add(a.getId());
+        }
+
+        if (!normalIds.isEmpty()) {
+            // action for normal minions - update salt master, the channels will be updated after minion restart
+            Map<String, Object> pillar = new HashMap<>();
+            pillar.put("mgr_server", proxy.map(p -> p.getHostname()).orElse(ConfigDefaults.get().getCobblerHost()));
+
+            Action a = scheduleApplyStates(loggedInUser, normalIds,
+                       Collections.singletonList(ApplyStatesEventMessage.SET_PROXY),
+                       Optional.of(pillar), new Date(), Optional.empty());
+            a = ActionFactory.save(a);
+            taskomaticApi.scheduleActionExecution(a);
```

Comment:
This can throw an exception. In this way, it will return an error, but if we have also ssh minion, they would be running ok but w will return an error.